File: ios/BaseiOSSwitUI/BaseiOSSwitUI/FeatureHome/View/HomeState.swift
//
//  HomeState.swift
//  BaseiOSSwitUI
//
//  Created by Ali Taha on 4.12.2023.
//

import Foundation

struct HomeState {
    var isLoading: Bool = true
    var error: String = ""
    var coins: [CoinHomeModel] = []
    
    func copy(
        isLoading: Bool? = nil,
        error: String? = nil,
        coins: [CoinHomeModel]? = nil
    ) -> HomeState {
        return HomeState(
            isLoading: isLoading ?? self.isLoading,
            error: error ?? self.error,
            coins: coins ?? self.coins
        )
    }
}

File: libraries/KMPKtorCoinbase/shared/src/commonMain/kotlin/dev/subfly/kmpktorcoinbase/featureHome/state/HomeState.kt
package dev.subfly.kmpktorcoinbase.featureHome.state

import dev.subfly.kmpktorcoinbase.core.model.CoinHomeModel
import dev.subfly.kmpktorcoinbase.core.util.Constants

data class HomeState(
    val coins: List<CoinHomeModel> = emptyList(),
    val isLoading: Boolean = true,
    val error: String = Constants.EMPTY_STRING,
)
